### PR TITLE
feat: set deploy URL as runtime property

### LINKED ISSUE #624: Sequential call causes wrong .js path
**Actual Behavior**

We see calls to the .js files being requested on sub directories and api endpoints. When using a browser, this doesn't seem to happen, bot search engines or other bots can cause this. 

**Expected Behavior**

The .js files exist on the root, so should always be called on root.

**Steps to Reproduce the Bug**

1. Checkout intershop-pwa
2. Build the site with docker with the default ICM_BASE_URL
2. Open the site via localhost, e.g. http://localhost:4300
3. Open one of the demo pages, e.g. http://localhost:4300/Acer/Acer-ICONIA-W500P-%2B-Keydock-sku10957684-catComputers.897.897_Acer?view=grid
4. Hit refresh several times. The log in the console should be 'normal' and contain .js files like
`GET /main.0e80287f29ca0cba7f99.js 200 330820 - 3.958 ms`
5. Next use a tool like siege to call the same URL. 
6. The first request to the page will show the same loggin, but when the second, third, etc visit is triggered, the path to the files change:
`siege -t5s -c1  http://localhost:4300/Acer/Acer-ICONIA-W500P-%2B-Keydock-sku10957684-catComputers.897.897_Acer\?view\=grid`
`SSR /Acer/main.0e80287f29ca0cba7f99.js`
4. Another example: siege -t5s -c1  http://localhost:4300/page/page.privacyPolicy.pagelet2-Page
`ERROR Http failure response for https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/cms/pages/main.0e80287f29ca0cba7f99.js: 404 Not Found`

**Environment Details**

- Desktop:

  - OS: Ubuntu 20.20
  - Docker in Terminal
  - 0.28.0

**Additional Context, like Screenshots, Log File Snippets etc.**

```
SSR /Acer/Acer-ICONIA-W500P-%2B-Keydock-sku10957684-catComputers.897.897_Acer?view=grid
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/products/10957684?allImages=true 200 19564 - 149 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/products/10957684/links 200 1520 - 232 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 361 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories/Computers/897/897_Acer 200 2244 - 374 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories?imageView=NO-IMAGE&view=tree&limit=2&omitHasOnlineProducts=true 200 130712 - 430 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/cms/includes/include.dialog.readyForShipment.pagelet2-Include 200 9402 - 287 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories/Computers/897 200 21072 - 224 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/products/10696946?allImages=true 200 19388 - 260 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/products/7918682?allImages=true 200 5882 - 316 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories/Computers 200 16760 - 300 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/products/5920217?allImages=true 200 19516 - 363 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/products/8899566?allImages=true 200 20582 - 373 ms
RES 200 /Acer/Acer-ICONIA-W500P-%2B-Keydock-sku10957684-catComputers.897.897_Acer?view=grid
GET /Acer/Acer-ICONIA-W500P-%2B-Keydock-sku10957684-catComputers.897.897_Acer?view=grid 200 207394 - 853.745 ms
SSR /Acer/main.0e80287f29ca0cba7f99.js
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 151 ms
RES 404 /Acer/main.0e80287f29ca0cba7f99.js
GET /Acer/main.0e80287f29ca0cba7f99.js 404 12667 - 492.240 ms
SSR /Acer/vendor.2dcf650556e821b23289.js
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 173 ms
RES 404 /Acer/vendor.2dcf650556e821b23289.js
GET /Acer/vendor.2dcf650556e821b23289.js 404 12673 - 483.965 ms
SSR /Acer/polyfills.efe1c9a4d455f9af5703.js
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 152 ms
RES 404 /Acer/polyfills.efe1c9a4d455f9af5703.js
GET /Acer/polyfills.efe1c9a4d455f9af5703.js 404 12682 - 478.675 ms
SSR /Acer/runtime.ee14c3e624fb5cfe0cb8.js
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 158 ms
RES 404 /Acer/runtime.ee14c3e624fb5cfe0cb8.js
GET /Acer/runtime.ee14c3e624fb5cfe0cb8.js 404 12676 - 480.909 ms
```

```
SSR /page/page.privacyPolicy.pagelet2-Page
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 240 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/cms/pages/page.privacyPolicy.pagelet2-Page 200 12626 - 254 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories?imageView=NO-IMAGE&view=tree&limit=2&omitHasOnlineProducts=true 200 130712 - 332 ms
RES 200 /page/page.privacyPolicy.pagelet2-Page
GET /page/page.privacyPolicy.pagelet2-Page 200 89641 - 531.319 ms
SSR /page/main.0e80287f29ca0cba7f99.js
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 221 ms
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories?imageView=NO-IMAGE&view=tree&limit=2&omitHasOnlineProducts=true 200 130712 - 280 ms
ERROR Http failure response for https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/cms/pages/main.0e80287f29ca0cba7f99.js: 404 Not Found
RES 200 /page/main.0e80287f29ca0cba7f99.js
GET /page/main.0e80287f29ca0cba7f99.js 200 80410 - 512.763 ms
SSR /page/main.0e80287f29ca0cba7f99.js
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/configurations 200 3818 - 171 ms
ERROR Http failure response for https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/cms/pages/main.0e80287f29ca0cba7f99.js: 404 Not Found
GET https://pwa-ish-demo.test.intershop.com/INTERSHOP/rest/WFS/inSPIRED-inTRONICS-Site/rest;loc=en_US;cur=USD/categories?imageView=NO-IMAGE&view=tree&limit=2&omitHasOnlineProducts=true 200 130712 - 309 ms
RES 200 /page/main.0e80287f29ca0cba7f99.js

```

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

[x] Feature

## What Is the Current Behavior?

Deploy URL in links to static resources is sometimes used relatively and sometimes absolute.

Issue Number: Closes #624

## What Is the New Behavior?

The deploy URL (which can be set for default Angular projects with [`ng build --deploy-url`](https://angular.io/cli/build)) can be set via environment variable `DEPLOY_URL` to the SSR process.

If not set, the default deployment URL `/` is applied, which ensures, all static assets are correctly loaded from the deployment root. 

This PR also makes it possible to use deeply nested `baseHref`s with a Multi-Channel context path setup.

This PR depends on #655 to make the Angular application independent on the `DEPLOY_URL`.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

## TODO

- [x] Documentation -> #646

### CODE REVIEW

**Comment by dhhyi** [comment]
I have to take this back to WIP as it breaks `ng serve`

**Comment by dhhyi** [comment]
wait for integration of #655

File: docs/concepts/deploy-url.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<!--
+kb_concepts
+kb_pwa
+kb_everyone
+kb_sync_latest_only
+-->
+
+# Deploy URL
```
Comment (jometzner) [inline docs/concepts/deploy-url.md:8]: @marschmidt89 We have a new documentation snippet here. I'll read through it semantically. Could you please check according to your standards?

**Review by marschmidt89 — APPROVED** [review]
documentation review done

**Comment by jometzner** [comment]
Closes AB#64439

**Review by jometzner — APPROVED** [review]

**Review by marschmidt89 — APPROVED** [review]
documentation review done

**Review by jometzner — APPROVED** [review]

**Pull request merged**